config/src/test/java/io/scalecube/config/MultimapConfigPropertyTest.java

```diff
@@ -0,0 +1,170 @@
+  }
+
+  @Test
+  public void testStringMultimapProperty() {
+    ImmutableMap<String, List<String>> expectedMultimap = ImmutableMap.<String, List<String>>builder()
+        .put("key1", ImmutableList.of("a"))
+        .put("key2", ImmutableList.of("b", "c", "d"))
+        .put("key3", ImmutableList.of("e"))
+        .build();
+    when(configSource.loadConfig())
+        .thenReturn(toConfigProps(mapBuilder()
+            .put("string_multimap", "key1=a,key2=b,c,d,key3=e")
+            .build()));
+
+    ConfigRegistryImpl configRegistry = newConfigRegistry(configSource);
+
+    assertEquals(expectedMultimap, configRegistry.stringMultimapProperty("string_multimap").valueOrThrow());
+  }
+
+  @Test
+  public void testMultimapPropertyWithDuplicatedKeys() {
+    ImmutableMap<String, List<Integer>> expectedMultimap = ImmutableMap.<String, List<Integer>>builder()
+        .put("key1", ImmutableList.of(1, 6))
+        .put("key2", ImmutableList.of(2, 3, 4, 7))
+        .put("key3", ImmutableList.of(5))
+        .build();
+    when(configSource.loadConfig())
+        .thenReturn(toConfigProps(mapBuilder()
+            .put("int_multimap", "key1=1,key2=2,3,4,key3=5,key1=6,key2=7")
+            .build()));
+
+    ConfigRegistryImpl configRegistry = newConfigRegistry(configSource);
+
+    assertEquals(expectedMultimap, configRegistry.intMultimapProperty("int_multimap").valueOrThrow());
+  }
+
+  @Test
+  public void testMultimapPropertyWithoutKeyValueSeparator() {
+    when(configSource.loadConfig())
+        .thenReturn(toConfigProps(mapBuilder()
+            .put("string_multimap", "value1")
+            .build()));
+
+    ConfigRegistryImpl configRegistry = newConfigRegistry(configSource);
+
+    assertEquals(Collections.emptyMap(), configRegistry.stringMultimapProperty("string_multimap").valueOrThrow());
+  }
+
+  @Test
+  public void testMultimapPropertyWithoutKey() {
+    ImmutableMap<String, List<String>> expectedMultimap = ImmutableMap.<String, List<String>>builder()
+        .put("", ImmutableList.of("value1"))
+        .build();
+    when(configSource.loadConfig())
+        .thenReturn(toConfigProps(mapBuilder()
+            .put("string_multimap", "=value1")
+            .build()));
+
+    ConfigRegistryImpl configRegistry = newConfigRegistry(configSource);
+
+    assertEquals(expectedMultimap, configRegistry.stringMultimapProperty("string_multimap").valueOrThrow());
+  }
+
+  @Test
+  public void testMultimapPropertyWithoutValue() {
+    ImmutableMap<String, List<String>> expectedMultimap = ImmutableMap.<String, List<String>>builder()
+        .put("key1", ImmutableList.of(""))
+        .build();
+    when(configSource.loadConfig())
+        .thenReturn(toConfigProps(mapBuilder()
+            .put("string_multimap", "key1=")
+            .build()));
+
+    ConfigRegistryImpl configRegistry = newConfigRegistry(configSource);
+
+    assertEquals(expectedMultimap, configRegistry.stringMultimapProperty("string_multimap").valueOrThrow());
+  }
+}
```

Comment:
Need additional tests:
1 multimap validator would applied if set
2 callback would be applied if set
3 declaring multimap as field in object config property should be allowed (see tests for reference: 
 `ObjectPropertyFieldTest.testListObjectPropertyField()`  and `ObjectPropertyFieldTest.testUntypedListNotSupported()`)

Reply:
The second case (callbacl) is difficult - all configProperties invoke with disable callbacks, expect for ConfigRegistryImpl.loadAndNotify()

Reply:
done